crates/matrix-sdk-ui/src/timeline/controller/mod.rs

```diff
@@ -1523,15 +1514,12 @@ impl<P: RoomDataProvider> TimelineController<P, (OlmMachine, OwnedRoomId)> {
     }
 }
 
-#[allow(clippy::too_many_arguments)]
 async fn fetch_replied_to_event(
     mut state: RwLockWriteGuard<'_, TimelineState>,
     index: usize,
     item: &EventTimelineItem,
     internal_id: TimelineUniqueId,
-    message: &Message,
-    reactions: &ReactionsByKeyBySender,
-    thread_root: &Option<OwnedEventId>,
+    aggregated: &AggregatedTimelineItemContent,
```

Comment:
I don't think you need this parameter either; `item` holds this aggregated content under the hood, right?

Reply:
Ah good point. We can either let it like this, or use an `.expect()` statement that the item is an aggregated item (and specify it clearly in this function's signature). But overall, since this function is only called once, I'm also tempted to suggest inlining it into its caller…

Reply:
I tried both but not quite happy with how it looked, I think I would like to just leave it alone for now if you don't mind 🤔